docs/src/index.md

```diff
@@ -1,160 +1,263 @@
-# Catalyst.jl for Reaction Network Modeling
+# [Catalyst.jl for Reaction Network Modeling](@id doc_index)
 
-Catalyst.jl is a symbolic modeling package for analysis and high performance
+Catalyst.jl is a symbolic modeling package for analysis and high-performance
 simulation of chemical reaction networks. Catalyst defines symbolic
-[`ReactionSystem`](@ref)s, which can be created programmatically or easily
-specified using Catalyst's domain specific language (DSL). Leveraging
-[ModelingToolkit.jl](https://docs.sciml.ai/ModelingToolkit/stable/) and
-[Symbolics.jl](https://docs.sciml.ai/Symbolics/stable/), Catalyst enables
+[`ReactionSystem`](@ref)s,
+which can be created programmatically or easily
+specified using Catalyst's domain-specific language (DSL). Leveraging
+[ModelingToolkit.jl](https://github.com/SciML/ModelingToolkit.jl) and
+[Symbolics.jl](https://github.com/JuliaSymbolics/Symbolics.jl), Catalyst enables
 large-scale simulations through auto-vectorization and parallelism. Symbolic
 `ReactionSystem`s can be used to generate ModelingToolkit-based models, allowing
 the easy simulation and parameter estimation of mass action ODE models, Chemical
 Langevin SDE models, stochastic chemical kinetics jump process models, and more.
 Generated models can be used with solvers throughout the broader
-[SciML](https://sciml.ai) ecosystem, including higher level SciML packages (e.g.
+[SciML](https://sciml.ai) ecosystem, including higher-level SciML packages (e.g.
 for sensitivity analysis, parameter estimation, machine learning applications,
 etc).
 
-## Features
-- A DSL provides a simple and readable format for manually specifying chemical
-  reactions.
-- Catalyst `ReactionSystem`s provide a symbolic representation of reaction networks,
-  built on [ModelingToolkit.jl](https://docs.sciml.ai/ModelingToolkit/stable/) and
-  [Symbolics.jl](https://docs.sciml.ai/Symbolics/stable/).
-- Non-integer (e.g. `Float64`) stoichiometric coefficients are supported for generating
-  ODE models, and symbolic expressions for stoichiometric coefficients are supported for
-  all system types.
-- The [Catalyst.jl API](@ref) provides functionality for extending networks,
-  building networks programmatically, network analysis, and for composing multiple
-  networks together.
-- `ReactionSystem`s generated by the DSL can be converted to a variety of
-  `ModelingToolkit.AbstractSystem`s, including symbolic ODE, SDE and jump process
-  representations.
-- Coupled differential and algebraic constraint equations can be included in
-  Catalyst models, and are incorporated during conversion to ODEs or steady
-  state equations.
-- Conservation laws can be detected and applied to reduce system sizes, and
-  generate non-singular Jacobians, during conversion to ODEs, SDEs, and steady
-  state equations.
+## [Features](@id doc_index_features)
```

Comment:
this and other sections with chanegs (except for one which is marked) is the same as the readme (but using proper refs and dynamic code)